lib/active_reporting/reporting_dimension.rb

```diff
@@ -101,14 +105,32 @@ def label_callback
 
     def determine_label_field(label_field)
       @label = if label_field.present? && validate_hierarchical_label(label_field)
-                 label_field.to_sym
+                 type == Dimension::TYPES[:degenerate] ? name : label_field.to_sym
+               elsif type == Dimension::TYPES[:degenerate]
+                 name
                else
                  dimension_fact_model.dimension_label || Configuration.default_dimension_label
                end
     end
 
     def determine_label_name(label_name)
-      @label_name = label_name ? "#{name}_#{label_name}" : name
+
+      if label_name
+        @label_name = label_name
+      else
+        @label_name = name
+        @label_name += "_#{@label}" if (type == Dimension::TYPES[:standard] && @label != :name)
```

Comment:
Technically these parentheses aren't needed.

Reply:
Technically yes, but for some reason i got an error at label name that i solved adding the parentheses. Can't explain way, but removing the parentheses will break the code/tests.